client/chain/chain.go

```diff
@@ -1260,6 +1279,32 @@ func (c *chainClient) StreamOrderbookUpdateEvents(orderbookType OrderbookType, m
 	}
 }
 
+// adjustGasPrices increases gas price to prioritize the tx for block inclusion
+func (c *chainClient) adjustGasPrices() {
+	var adjustedPrices string
+	for _, price := range c.txFactory.GasPrices() {
+		if adjustedPrices != "" {
+			adjustedPrices += ","
+		}
+		amount := price.Amount.Mul(c.gasPriceIncrement).RoundInt().String()
+		denom := price.Denom
+		adjustedPrices += amount + denom
+	}
+	c.txFactory = c.txFactory.WithGasPrices(adjustedPrices)
+	c.logger.Infoln("gas price adjusted to", c.txFactory.GasPrices())
+}
+
+// resetGasPrices resets gas price to default value to avoid wasting gas when it's network is not busy
+func (c *chainClient) resetGasPrices(defaultGasPrices string, timer *time.Ticker) {
```

Comment:
We should have a ResetGasPrice func to explicit reset gas. It's helpful when the bot realize that all txs has passed so it don't need adjusted gas price anymore, this go routine is not controllable